mutiny-core/src/nostr/mod.rs

```diff
@@ -117,11 +135,27 @@ impl<S: MutinyStorage> NostrManager<S> {
     /// Creates a new NWC profile and saves to storage
     pub(crate) fn create_new_profile(
         &self,
-        name: String,
+        profile_type: ProfileType,
         max_single_amt_sats: u64,
     ) -> Result<NwcProfile, MutinyError> {
         let mut profiles = self.nwc.write().unwrap();
-        let index = profiles.len() as u32;
+
+        let normal_profiles_count = profiles
+            .iter()
+            .filter(|&nwc| nwc.profile.index >= USER_NWC_PROFILE_START_INDEX)
+            .count() as u32;
+
+        let (name, index) = match profile_type {
+            ProfileType::Reserved(reserved_profile) => {
+                let (name, index) = reserved_profile.info();
+                (name.to_string(), index)
+            }
+            // Ensure normal profiles start from 100
+            ProfileType::Normal { name } => {
+                (name, normal_profiles_count + USER_NWC_PROFILE_START_INDEX)
```

Comment:
should put a check here that the index isn't over 1000, not sure what we should do in the case that happens

Reply:
maybe we could do negative numbers instead

Reply:
I now set a panic where we convert `ReservedProfile` to an index. So it'll never have an index that is greater than 1000.

Reply:
And normal profile counts *can* be over 1000. Not sure if that's what you were referring to now that I read your comment again.